client/src/tests/modules/auth/container/index.test.jsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { shallow } from "enzyme";
+import { SignIn, mapStateToProps } from "../../../../modules/Auth/container";
+import { props, responses, expected } from "../authMockData";
+
+describe("<Testing the redux container />", () => {
+  it("should mount without crashing", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+
+  it("should test for returned errors on authentication ", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().componentWillReceiveProps(props.containerPropsErrors);
+    const actual = wrapper.instance().state;
+    expect(actual).toEqual(expected.notSuccessful);
+  });
+
+
+  it("should test for returned messages on authentication ", () => {
+    const wrapper = shallow(<SignIn {...props.containerProps} />);
+    wrapper.instance().componentWillReceiveProps(props.containerPropsMessage);
```

Comment:
what do you think about using the `wrapper.setProps(nextProps: object)` function to test componentWillReceiveProps as opposed to calling it directly like this

Reply:
Nice, I did not think about it in that aspect, I guess we will still achieve the same thing either way.